awx/main/tasks/receptor.py

```diff
@@ -258,157 +258,132 @@ class AWXReceptorJob:
     def __init__(self, task, runner_params=None):
         self.task = task
         self.runner_params = runner_params
-        self.unit_id = None
+        # We establish a connection to the Receptor socket
+        self.receptor_ctl = get_receptor_ctl()
 
         if self.task and not self.task.instance.is_container_group_task:
             execution_environment_params = self.task.build_execution_environment_params(self.task.instance, runner_params['private_data_dir'])
             self.runner_params.update(execution_environment_params)
 
+        self.work_type = self.get_work_type()
+
         if not settings.IS_K8S and self.work_type == 'local' and 'only_transmit_kwargs' not in self.runner_params:
             self.runner_params['only_transmit_kwargs'] = True
 
     def run(self):
-        # We establish a connection to the Receptor socket
-        receptor_ctl = get_receptor_ctl()
-
         res = None
+        unit_id = None
         try:
-            res = self._run_internal(receptor_ctl)
+            res, unit_id = self._run()
+
+            if res.status == 'error':
+                self.handle_error(unit_id)
```

Comment:
@shanemcd I felt uneasy about this before, and putting the change up in this form, I don't think it's acceptable.

The error handling code in this version has no watcher, so it won't respond to SIGTERM events. This was the case before... but when the cancel rework I was doing, it _is_ actually solved as long as it's in the main try-except, the cancel watching will be active even if the error handling code is blocked on a read.

The solution shouldn't be hard, but it is just to move this back into the main thread-pool block. It's just a TODO, and I want to note it.